pkg/didmethod/peer/did.go

```diff
@@ -64,10 +64,8 @@ func computeDid(doc *did.Doc) (string, error) {
 func validateDID(doc *did.Doc) error {
 	peerDid := doc.ID
 
-	matched, err := regexp.MatchString(`did:peer:11-([a-fA-F0-9]){64}`, peerDid)
-	if err != nil {
-		return fmt.Errorf("regex match string failed %w", err)
-	}
+	didRegex := regexp.MustCompile(`did:peer:11-([a-fA-F0-9]){64}`)
```

Comment:
move it out of function body

Reply:
@jessie-codes Thanks for your contribution. @soluchok suggests that the compiled regular expression is moved out of the function body so that it doesn't need to be recompiled on every call.

Reply:
@soluchok, your suggestion definitely makes sense; however, since there is a linting rule for no global variables, there isn't a great place for this to outside of the function. I can't make it a const due to it not being a simple value and this function isn't attached to a struct.

Given those constraints, is there a preference on where you would want the variable for the regex to be defined?

Reply:
@jessie-codes you may add an exception for this case using a nolint comment. there is an example here: https://github.com/hyperledger/aries-framework-go/blob/9582e05f3a7adcb583469da0a0ed074487c9aa22/pkg/doc/verifiable/jwt_unsecured_test.go#L17

Reply:
Great, I've updated the branch with the change.